Source/Core/VideoBackends/OGL/Render.cpp

```diff
@@ -1342,110 +1342,80 @@ void Renderer::SwapImpl(u32 xfbAddr, u32 fbWidth, u32 fbStride, u32 fbHeight, co
 
 	// Frame dumps are handled a little differently in Windows
 	// Frame dumping disabled entirely on GLES3
-	if (GLInterface->GetMode() == GLInterfaceMode::MODE_OPENGL)
-	{
 #if defined _WIN32 || defined HAVE_LIBAV
-		if (SConfig::GetInstance().m_DumpFrames)
-		{
-			std::lock_guard<std::mutex> lk(s_criticalScreenshot);
-			if (frame_data.empty() || w != flipped_trc.GetWidth() ||
-				     h != flipped_trc.GetHeight())
-			{
-				w = flipped_trc.GetWidth();
-				h = flipped_trc.GetHeight();
-				frame_data.resize(3 * w * h);
-			}
-			glPixelStorei(GL_PACK_ALIGNMENT, 1);
-			glReadPixels(flipped_trc.left, flipped_trc.bottom, w, h, GL_BGR, GL_UNSIGNED_BYTE, &frame_data[0]);
-			if (w > 0 && h > 0)
-			{
-				if (!bLastFrameDumped)
-				{
-					#ifdef _WIN32
-						bAVIDumping = AVIDump::Start(nullptr, w, h);
-					#else
-						bAVIDumping = AVIDump::Start(w, h);
-					#endif
-					if (!bAVIDumping)
-					{
-						OSD::AddMessage("AVIDump Start failed", 2000);
-					}
-					else
-					{
-						OSD::AddMessage(StringFromFormat(
-									"Dumping Frames to \"%sframedump0.avi\" (%dx%d RGB24)",
-									File::GetUserPath(D_DUMPFRAMES_IDX).c_str(), w, h), 2000);
-					}
-				}
-				if (bAVIDumping)
-				{
-					#ifndef _WIN32
-						FlipImageData(&frame_data[0], w, h);
-					#endif
-
-						AVIDump::AddFrame(&frame_data[0], w, h);
-				}
+	if (SConfig::GetInstance().m_DumpFrames)
+	{
+		std::lock_guard<std::mutex> lk(s_criticalScreenshot);
+		int bytes_per_pixel = 3;
+		GLenum data_type = GL_BGR;
+		AVIDump::DumpFormat dump_fmt = AVIDump::DumpFormat::FORMAT_BGR;
 
-				bLastFrameDumped = true;
-			}
-			else
-			{
-				NOTICE_LOG(VIDEO, "Error reading framebuffer");
-			}
-		}
-		else
+		if (GLInterface->GetMode() == GLInterfaceMode::MODE_OPENGLES3)
 		{
-			if (bLastFrameDumped && bAVIDumping)
-			{
-				std::vector<u8>().swap(frame_data);
-				w = h = 0;
-				AVIDump::Stop();
-				bAVIDumping = false;
-				OSD::AddMessage("Stop dumping frames", 2000);
-			}
-			bLastFrameDumped = false;
+			bytes_per_pixel = 4;
+			data_type = GL_RGBA;
+			dump_fmt = AVIDump::DumpFormat::FORMAT_RGBA;
 		}
-#else
-		if (SConfig::GetInstance().m_DumpFrames)
+
+		if (frame_data.empty() ||
+		    w != flipped_trc.GetWidth() || h != flipped_trc.GetHeight())
```

Comment:
A no-op resizing call is very cheap. so there is no need to have a global state for w and h. Just always call resize.